lib/withLoggedInUser.js

```diff
@@ -101,8 +101,14 @@ const withLoggedInUser = WrappedComponent => {
           newToken = await refreshTokenWithTwoFactorCode(token, twoFactorAuthenticatorCode);
           setLocalStorage(LOCAL_STORAGE_KEYS.ACCESS_TOKEN, newToken);
         } else if (recoveryCode) {
-          newToken = await refreshTokenWithTwoFactorRecoveryCode(token, recoveryCode);
+          const response = await refreshTokenWithTwoFactorRecoveryCode(token, recoveryCode);
+
+          newToken = response.token;
           setLocalStorage(LOCAL_STORAGE_KEYS.ACCESS_TOKEN, newToken);
+          // Redirect user
+          if (response.slug) {
+            window.location.href = `/${response.slug}/edit/two-factor-auth`;
+          }
```

Comment:
Can we think about another place where to handle the redirect?

That might work but getLoggedInUser should not have side effect like this.